src/eyeTracker.m

```diff
@@ -55,161 +30,134 @@
                     Eyelink('Shutdown');
-                    Screen('CloseAll');
+                    CleanUp();
                     return
                 end
 
-                % Make sure EL is still connected: returns 1 if connected,
-                % -1 if dummy-connected,
-                % 2 if broadcast-connected and 0 if not connected
-                ELconnection = Eyelink('IsConnected');
+                % Make sure EL is still connected: returns 1 if connected, -1 if dummy-connected,
+                %  2 if broadcast-connected and 0 if not connected. Exit program if this fails.
+                elConnection = Eyelink('IsConnected');
                 if ELconnection ~= 1
                     fprintf('Eyelink is not connected, aborted.\n');
                     Eyelink('Shutdown');
-                    Screen('CloseAll');
+                    CleanUp();
                     return
                 end
 
-                % Last check that the EL is up to work
+                % Last check that the EL is up to work and exit program if this fails.
                 if ~EyelinkInit(0, 1)
                     fprintf('Eyelink Init aborted.\n');
+                    CleanUp();
                     return
                 end
 
-                % Open the edf file to write the data
+                % Open the edf file to write the data.
                 edfFile = 'demo.edf';
                 Eyelink('Openfile', edfFile);
 
+                % Get EyeLink setup information.
                 [el.v, el.vs] = Eyelink('GetTrackerVersion');
                 fprintf('Running experiment on a ''%s'' tracker.\n', el.vs);
 
-                % make sure that we get gaze data from the Eyelink
+                % Make sure that we get gaze data from the Eyelink.
                 Eyelink('Command', 'link_sample_data = LEFT,RIGHT,GAZE,AREA');
+                Eyelink('command', 'link_event_data = GAZE,GAZERES,HREF,AREA,VELOCITY');
+                Eyelink('command', 'link_event_filter = LEFT,RIGHT,FIXATION,BLINK,SACCADE,BUTTON');
 
-                %% STEP 4
-                % SET UP TRACKER CONFIGURATION
-
-                % Setting the proper recording resolution, proper calibration type,
-                % as well as the data file content;
-
-                Eyelink('command', 'add_file_preamble_text', ...
-                    'Recorded by EyelinkToolbox demo-experiment');
+                %% Calibration
 
                 % This command is crucial to map the gaze positions from the tracker to
-                % screen pixel positions to determine fixation
+                %  screen pixel positions to determine fixation.
                 Eyelink('command', 'screen_pixel_coords = %ld %ld %ld %ld', 0, 0, 0, 0);
                 Eyelink('message', 'DISPLAY_COORDS %ld %ld %ld %ld', 0, 0, 0, 0);
 
+                % Set calibration type.
+                Eyelink('command', 'calibration_type = HV5');
```

Comment:
I think that we want to be able to define the calibration type from setParameters.
- HV5: horizontal ; vertical ; 5 points
- H3 for horizontal calibration at 3 positions
- HV9 for horizontal and vertical calibration at 9 positions.

We need to see what calibration type eyelink offers by default.

https://docs.google.com/document/d/1eggzTCzSHG3AEKhtnEDbcdk-2avXN6I94X8aUPEBVsw/edit?disco=AAAAB8R3onE

Reply:
ok, will do it